cores/rp2040/SerialUART.cpp

```diff
@@ -365,6 +366,26 @@ SerialUART::operator bool() {
     return _running;
 }
 
+void SerialUART::setBreakReceived() {
+    CoreMutex m(&_mutex);
+    if (!_running || !m) {
+        return;
+    }
+
+    _break = true;
+}
+
+bool SerialUART::getBreakReceived() {
+    CoreMutex m(&_mutex);
+    if (!_running || !m) {
```

Comment:
Also, this should probably be the `_fifoMutex` and not the global object one.  The reason is that this flag is essentially now an extension of the object's FIFO.  OTW there is a potential race condition between lines 384 and 385...

Reply:
Also, since it's driven off the UART FIFO it should do the same as `::peek` and `::read` to make sure to pump out any chars from the HW FIFO to the SW one (which is where we'd detect the BREAK).

i.e. 
````
    if (_polling) {
        _handleIRQ(false);
    } else {
        _pumpFIFO();
    }
````

OTW I can see a case where the IRQ doesn't actually pump the FIFO because the SW is doing some Serial call that disallows it, leaving the BREAK stuck in the HW side and never detected.

Sorry. BTW, for the piecemeal notes.  The concurrency w/the UARTs is kind of subtle, especially in multi-core, multi-threaded stuff like we have here!

Reply:
Good points, both.   I will take a look at redoing this later today.